wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/RecipeFileContentProvider.java

```
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.devfile.server;
+
+/**
+ * Fetches content of file described in local field of recipe-type {@link
+ * org.eclipse.che.api.devfile.model.Tool}
+ *
+ * @author Max Shaposhnyk
+ */
+@FunctionalInterface
+public interface RecipeFileContentProvider {
+
+  /**
+   * Fetches content of the file specified under 'local' field fo recipe-type tools in {@link
+   * org.eclipse.che.api.devfile.model.Devfile}.
+   *
+   * @param localFileName file name to fetch content. Only devfile-relative files are currently
+   *     supported, so it means file should be localed at the same directory level as devfile (no
+   *     matter in repository or PR or branch etc )
+   * @return content of the specified file or {@code null} if it is impossible to fetch it.
+   */
+  String fetchContent(String localFileName);
```

I think it would be better to throw an exception here if an exception occurred during file content fetching. It would allow propagating original exception (`https://raw.github/....repo/folder/petclinic.yaml is unreachable` or `invalid URL https://raw.github/....repo/folder/$#@#$%.yaml`) to a user and would make issues investigation easier.
I understand that it may be not so easy to implement since `URLFetcher` is designed in a way not to propagate exception but please consider implementing it if it makes sense for you.
Note my errors examples are imaginary, not sure if like the second one can happen.

---

with IOException shouldn't be very complex